man/coap_block.txt.in

```
@@ -337,6 +367,31 @@ body dropped.
 The *coap_q_block_is_supported*() function is used to determine whether
 libcoap has been build with Q-Block support or not.
 
+*Function: coap_register_block_data_handler()*
+
+The *coap_register_block_data_handler*() function is used to register a handler
+function that will be called with each received data block if
+COAP_BLOCK_SINGLE_BODY is enabled. This is especially relevant for cases where
+COAP_BLOCK_SINGLE_BODY is enabled implicitly, e.g., when COAP_BLOCK_TRY_Q_BLOCK
+is enabled, and the application wants to process the single data blocks
+nonetheless but without handling the complicated protocol signaling itself.
+
+The handler parameter _resource_ is only set for server-side reception of data
+and it is NULL for client-side reception. _body_data_ can be used to call
+*coap_block_build_body*() in the handler and let libcoap gather all data, e.g.,
+if the single blocks are only of interest in specific cases. It can also be
+used to store a pointer to memory allocated by the application that will later
+be presented to the method or response handler after the transmission finished
+eventually.
+
+*NOTE*: the corresponding method or response handler will still be called when
+the transmission is complete. If _body_handler_ was not set by the block data
+handler, the method or response handler will be called with _data_ parameter
+set to NULL.
+
+*NOTE*: the block data handler will be silently ignored for (reverse) proxy
+resources.
```

*NOTE*: the call to the block data handler will be silently ignored for (forward | reverse) proxy
resources.